### DISCUSSION TITLE
convert the features (bits vectors) back into a different representation

### DISCUSSION THREAD

**Discussion by andresilvapimentel** [rdkit/rdkit#5537] [Q&A]
I am using RDkit and DeepChem to understand the features importances of some datasets.

I splitted the dataset and featurized it using ECFP (in DeepChem) so that the features are stored in train_dataset.X object, which has the shape (1631, 1024). So, it has 1024 features as bits vectors.

Then, I created a model using Random Forest and generate the features importances.  Finally, I plotted the features against relative importance. However, as the features are bits vectors, the look of y-axis is weird using matplotlib library.

Is there any way to convert these features (bits vectors) into a different representation (smiles of fragments, per example)? How can I code this transformation?

I would be grateful if you can answer this question using RDkit or any other package.

Thank you in advance.

**Comment by andresilvapimentel** [comment]
I found this link:
https://www.rdkit.org/docs/GettingStartedInPython.html#explaining-bits-from-morgan-fingerprints

But it is not exactly what I want @greglandrum 

As I do not have a molecule to print the smile from the bits vector, it does not work for me. It would be better if I could get the feature  number (or other representation) using the bits vector.

**Reply by andresilvapimentel** [reply]
@greglandrum After running this piece of code:

importances = model.feature_importances_
indices = np.argsort(importances)
features = train_dataset.X
plt.title('Feature Importances')
plt.barh(range(len(indices)), importances[indices], color='b', align='center')
plt.yticks(range(len(indices)), [features[i] for i in indices])
plt.xlabel('Relative Importance')
plt.show()

I am getting something like this:

![IMG-20220905-WA0010](https://user-images.githubusercontent.com/79097989/188703921-706093cc-9461-48b0-b8f4-7886d4d9985c.jpg)

What I need is to change the representation of bits vector in features[i].
Did you get it? How can I change this representation?
